azext_iot/iothub/providers/state.py

```diff
@@ -51,7 +56,15 @@ def __init__(
             login=login,
             auth_type_dataplane=auth_type_dataplane
         )
+
+        if not login:
```

Comment:
its easier on the eyes if you swap the if and else (so do a if login, .... else ....)